builtin/logical/ssh/path_sign.go

```diff
@@ -275,14 +274,20 @@ func (b *backend) calculateKeyId(data *framework.FieldData, req *logical.Request
 		return reqId, nil
 	}
 
-	keyHash := sha256.Sum256(pubKey.Marshal())
-	keyId := hex.EncodeToString(keyHash[:])
+	keyIdFormat := "vault-{{token_display_name}}-{{public_key_hash}}"
+	if req.DisplayName == "" {
+		keyIdFormat = "vault-{{public_key_hash}}"
+	}
 
-	if req.DisplayName != "" {
-		keyId = fmt.Sprintf("%s-%s", req.DisplayName, keyId)
+	if role.KeyIDFormat != "" {
+		keyIdFormat = role.KeyIDFormat
 	}
 
-	keyId = fmt.Sprintf("vault-%s", keyId)
+	keyId := substQuery(keyIdFormat, map[string]string{
+		"token_display_name": req.DisplayName,
```

Comment:
What happens when role's `KeyIDFormat` mentions `{{token_display_name}}` and `req.DisplayName` is empty? I think we'll end up having a `{{}}` substring in `keyId`. Given that the role specified it, it seems reasonable to have an `{{}}` block. Regardless, can we have a comment here explaining that?